### PR TITLE
Ut store

### PR DESCRIPTION
**Pull request by vishala-r**
This PR implements unit tests for the store package. it covers 95.3% of the statements.

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -30,5 +30,11 @@ jobs:
     - name: Build
       run: make
 
+    - name: Change Permissions
+      run: |
+       sudo mkdir -p /var/lib/cni/inframanager
+       sudo chmod 777 -R /var/lib/cni/inframanager
```
Comment (venkateshpnv) [inline .github/workflows/build-and-test.yml:44]: Any reason why the permissions should be given to the group and others? I suggest to use 600 unless you have a requirement.

Comment (nupurjai) [inline .github/workflows/build-and-test.yml:44]: Please change it to 600 or 644. We do not want others to modify anything

File: pkg/inframanager/store/storeendpoint.go [hunk 2]
```diff
@@ -112,15 +149,15 @@ func (ep EndPoint) UpdateToStore() bool {
...
 func RunSyncEndPointInfo() bool {
-	jsonStr, err := json.MarshalIndent(EndPointSet.EndPointMap, "", " ")
+	jsonStr, err := JsonMarshalIndent(EndPointSet.EndPointMap, "", " ")
 	if err != nil {
 		log.Errorf("Failed to marshal endpoint entries map %s", err)
 		return false
 	}
 
-	if err = ioutil.WriteFile(storeFile, jsonStr, 0600); err != nil {
+	if err = NewWriteFile(StoreEpFile, jsonStr, 0600); err != nil {
```
Comment (nupurjai) [inline pkg/inframanager/store/storeendpoint.go:158]: This is fine

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (nupurjai) [reply .github/workflows/build-and-test.yml:44]: We can directly do this on runner

**Review by nupurjai — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (vishala-r) [reply .github/workflows/build-and-test.yml:44]: with 644, build is failing

**Review by venkateshpnv — APPROVED** [review]

**Pull request closed without merging**